Source/Blazorise/Components/Offcanvas/Offcanvas.razor.cs

```diff
@@ -0,0 +1,484 @@
+﻿#region Using directives
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Blazorise.Modules;
+using Blazorise.States;
+using Blazorise.Components;
+using Blazorise.Utilities;
+using Microsoft.AspNetCore.Components;
+using Microsoft.JSInterop;
+using Blazorise.Providers;
+using System.Runtime.CompilerServices;
+#endregion
+
+namespace Blazorise;
+
+/// <summary>
+/// A sidebar component that slides in and out of the screen.
+/// </summary>
+public partial class Offcanvas : BaseComponent, ICloseActivator, IAnimatedComponent, IHideableComponent, IAsyncDisposable
+{
+    #region Members
+
+    ///<summary>
+    /// Holds the state of the Offcanvas component.
+    ///</summary>
+    private OffcanvasState state = new()
+    {
+        Visible = false,
+    };
+
+    ///<summary>
+    /// Holds the reason for the Offcanvas closing.
+    ///</summary>
+    private CloseReason closeReason = CloseReason.None;
+
+    ///<summary>
+    /// Indicates whether the Offcanvas has been registered with JavaScript.
+    ///</summary>
+    private bool jsRegistered;
+
+    ///<summary>
+    /// Reference to the .NET object adapter for the CloseActivator.
+    ///</summary>
+    private DotNetObjectReference<CloseActivatorAdapter> dotNetObjectRef;
+
+    ///<summary>
+    /// List of element IDs that act as close activators for the Offcanvas.
+    ///</summary>
+    private readonly List<string> closeActivatorElementIds = new();
+
+    ///<summary>
+    /// Adapter for the Closeable interface to handle close events.
+    ///</summary>
+    private CloseableAdapter closeableAdapter;
+
+    ///<summary>
+    /// Event that is triggered when the Offcanvas is opened.
+    ///</summary>
+    internal event Action _Opened;
+
+    ///<summary>
+    /// Event that is triggered when the Offcanvas is closed.
+    ///</summary>
+    internal event Action _Closed;
```

Comment:
These are kinda specific. It shows on Modal also. They are only used internally, and since we already have a public parameter named `Opened`, and `Closes` I marked these internals with an `_`. Also, we already name internal components with `_`. I guess this is fine.

Reply:
hmm we do? I'll have to update docs later.

Reply:
Ah... an **internal componen**t is not the same as an **internal field**. Anyway if you deem it ok i'll update the covnentions later.

Reply:
> Anyway if you deem it ok i'll update the covnentions later.

You can. But be aware that is only when names are clashing with existing methods. usually, I always prefer it without `_`.

Reply:
That's one of the big advantages of having private stuff named with _ and why I like it :) no problems with public api collision and easy discoverability.